src/de.rs

```diff
@@ -434,6 +436,7 @@ struct DeserializerFromEvents<'de, 'document> {
     path: Path<'document>,
     remaining_depth: u8,
     current_enum: Option<CurrentEnum<'document>>,
+    is_serde_content_newtype: bool,
```

Comment:
I would add a doc comment describing what problem it is solving and why that way.